grub-core/lib/gpt.c

```diff
@@ -237,6 +249,64 @@ grub_gpt_header_check (struct grub_gpt_header *gpt,
 }
 
 static grub_err_t
+grub_gpt_check_primary (grub_gpt_t gpt)
```

Comment:
The two new check_backup and check_primary functions make more sense in the next PR after this one where I add additional tests that are specific to primary and backup headers and therefore cannot be shared in the header_check function.